### PR TITLE
feat: Add Rect2DOperationView option in PointCloud2DView

### PR DESCRIPTION
**Pull request by lihqi**

### CODE REVIEW

File: packages/lb-components/src/App.tsx [hunk 1]
```diff
@@ -210,7 +212,12 @@ const App: React.FC<AppProps> = (props) => {
   return (
     <div>
       <ConfigProvider locale={i18n.language === 'en' ? enUS : zhCN}>
-        <MainView {...props} intelligentFit={intelligentFit} />
+        <MainView
+          {...props}
+          intelligentFit={intelligentFit}
+          cuboidBoxIn2DView={cuboidBoxIn2DView}
+          checkMode={checkMode}
```
Comment (lihqi) [inline packages/lb-components/src/App.tsx:219]: 之前都没有传这个checkMode，加上会不会有问题

File: packages/lb-annotation/src/core/toolOperation/rectOperation.ts [hunk 2]
```diff
@@ -1210,6 +1208,7 @@ class RectOperation extends BasicToolOperation {
 
       // 同步 rectList
       this.emit('updateResult');
+      this.emit('updateDragResult', { ...this.selectedRect });
```
Comment (Glenfiddish) [inline packages/lb-annotation/src/core/toolOperation/rectOperation.ts:1211]: use new events should defined before

File: packages/lb-annotation/src/core/toolOperation/rect2DOperation.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import RectOperation from './rectOperation';
+
+class Rect2DOperation extends RectOperation {
```
Comment (Kerwin-L) [inline packages/lb-annotation/src/core/toolOperation/rect2DOperation.ts:3]: Please rename the classname. Rect2DOperation is same as RectOperation. And write more comment about this

File: packages/lb-components/src/App.tsx (hunk 1 above)
Comment (Kerwin-L) [reply packages/lb-components/src/App.tsx:219]: The `props` have `checkMode`  and `cuboidBoxIn2DView`. And it is defined in `AppProps`

File: packages/lb-components/src/components/rect2DOperationView/index.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import { a2MapStateToProps } from '@/store/annotation/map';
+import { LabelBeeContext } from '@/store/ctx';
+import { ImgUtils, Rect2DOperation, uuid } from '@labelbee/lb-annotation';
+import { IBasicRect } from '@labelbee/lb-utils';
+
+import { usePointCloudViews } from '../pointCloudView/hooks/usePointCloudViews';
+import { PointCloudContext } from '../pointCloudView/PointCloudContext';
+
+interface IRect2DOperationViewProps {
+  mappingData?: any;
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/rect2DOperationView/index.tsx:13]: Need to defined it

File: packages/lb-components/src/components/pointCloudView/PointCloud2DView.tsx [hunk 5]
```diff
@@ -183,21 +191,25 @@ const PointCloud2DView = ({ currentData, config }: IA2MapStateProps) => {
...
+            }}
+            afterImgOnLoad={afterImgOnLoad}
+            zoomInfo={{
+              min: 0.01,
+              max: 1000,
+              ratio: 0.4,
+            }}
+          />
+        ) : (
+          <Rect2DOperationView size={size} mappingData={mappingData} checkMode={checkMode} />
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/pointCloudView/PointCloud2DView.tsx:211]: `Rect2DOperationView` is totally different from `PointCloud2DView`. The logic in `PointCloud2DView` is not need to use in `Rect2DOperationView`

File: packages/lb-components/src/components/rect2DOperationView/index.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+const Rect2DOperationView = (props: IRect2DOperationViewProps) => {
+  const { mappingData = {}, size, config, checkMode } = props;
+  const { url } = mappingData;
+  const { pointCloudBoxList, setPointCloudResult, selectedID, setCuboidBoxIn2DView } =
+    useContext(PointCloudContext);
+
+  const { update2DViewRect } = usePointCloudViews();
+  const ref = React.useRef(null);
+  const operation = useRef<any>(null);
+  const currentFn = useRef<any>(null);
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/rect2DOperationView/index.tsx:38]: Use `useLatest`(ahooks) to replace it.

File: packages/lb-components/src/components/rect2DOperationView/index.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+  useEffect(() => {
+    currentFn.current = update2DViewRect;
+  }, [update2DViewRect]);
+
+  const setRects = () => {
+    let allRects: IRect2DOperationViewRect[] = [];
+    pointCloudBoxList.forEach((pointCloudBox) => {
+      const { rects = [], id, attribute, trackID } = pointCloudBox;
+      const rect = rects.find((rect) => rect.imageName === mappingData.url);
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/rect2DOperationView/index.tsx:53]: You can define more Info in `IMappingImg`. 

Like add `path` in `IMappingImg`.
image

This way is update in `sensebee`
image

File: packages/lb-components/src/components/rect2DOperationView/index.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        config: { ...config, isShowOrder: true, attributeConfigurable: true },
+        checkMode,
+      });
+
+      operation.current = toolInstance;
+      operation.current.init();
+      operation.current.on('updateDragResult', handleUpdateDragResult);
+
+      return () => {
+        operation.current?.unbindAll('updateDragResult');
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/rect2DOperationView/index.tsx:76]: `operation.current?.unbind('updateDragResult',  handleUpdateDragResult) ` is better way

File: packages/lb-components/src/components/rect2DOperationView/index.tsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    currentFn.current = update2DViewRect;
+  }, [update2DViewRect]);
+
+  const setRects = () => {
+    let allRects: IRect2DOperationViewRect[] = [];
+    pointCloudBoxList.forEach((pointCloudBox) => {
+      const { rects = [], id, attribute, trackID } = pointCloudBox;
+      const rect = rects.find((rect) => rect.imageName === mappingData.url);
+      if (rect) {
+        allRects = [...allRects, { ...rect, boxID: id, id: uuid(), attribute, order: trackID }];
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/rect2DOperationView/index.tsx:55]: You need to think that if the pointCloudList render, the `id` is always different from before.

File: packages/lb-components/src/utils/index.ts [hunk 10]
```diff
@@ -130,10 +130,10 @@ export const getRectPointCloudBox = async (
     ...getBoundingRect(tmpPoints),
     valid,
     trackID,
-    imageName: mappingData.url,
+    path: mappingData.path,
```
Comment (Kerwin-L) [inline packages/lb-components/src/utils/index.ts:133]: The annotation results still need to be harmonized with the client's

The `path` need to changed to `imageName`

File: packages/lb-utils/src/types/pointCloud.ts [hunk 11]
```diff
@@ -79,7 +79,7 @@ export interface I3DSpaceCoord {
 export interface IPointCloudBoxRect extends IBasicRect {
   valid: boolean;
   trackID: number;
-  imageName: string;
+  path: string;
```
Comment (Kerwin-L) [inline packages/lb-utils/src/types/pointCloud.ts:82]: Same as comment above

File: packages/lb-components/src/views/MainView/toolFooter/SwitchCuboidBoxIn2DView/index.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  return (
+    <>
+      <span
+        style={{
+          marginRight: 10,
+        }}
+      >
+        图片标注物
```
Comment (Kerwin-L) [inline packages/lb-components/src/views/MainView/toolFooter/SwitchCuboidBoxIn2DView/index.tsx:28]: The text needs to be internationalized.

**Pull request merged**